sdk/iot/provisioning/src/az_iot_provisioning_client.c

```diff
@@ -47,161 +54,420 @@ AZ_NODISCARD az_result az_iot_provisioning_client_init(
+  }
+
+  // Parse the status.
+  az_span remainder = az_span_slice(
+      received_topic, az_span_size(str_dps_registrations_res), az_span_size(received_topic));
+
+  az_span int_slice = az_span_token(remainder, AZ_SPAN_FROM_STR("/"), &remainder);
+  AZ_RETURN_IF_FAILED(az_span_atou32(int_slice, (uint32_t*)(&out_response->status)));
+
+  // Parse the optional retry-after= field.
+  az_span retry_after = AZ_SPAN_FROM_STR("retry-after=");
+  idx = az_span_find(remainder, retry_after);
+  if (idx != -1)
+  {
+    remainder = az_span_slice(remainder, idx + az_span_size(retry_after), az_span_size(remainder));
+    int_slice = az_span_token(remainder, AZ_SPAN_FROM_STR("&"), &remainder);
+
+    AZ_RETURN_IF_FAILED(az_span_atou32(int_slice, &out_response->retry_after_seconds));
+  }
+  else
+  {
+    out_response->retry_after_seconds = 0;
+  }
+
+  AZ_RETURN_IF_FAILED(az_iot_provisioning_client_parse_payload(received_payload, out_response));
+
+  return AZ_OK;
 }
 
 // $dps/registrations/PUT/iotdps-register/?$rid=%s
-AZ_NODISCARD az_result az_iot_provisioning_client_register_publish_topic_get(
+AZ_NODISCARD az_result az_iot_provisioning_client_register_get_publish_topic(
     az_iot_provisioning_client const* client,
-    az_span mqtt_topic,
-    az_span* out_mqtt_topic)
+    char* mqtt_topic,
+    size_t mqtt_topic_size,
+    size_t* out_mqtt_topic_length)
 {
   (void)client;
 
   AZ_PRECONDITION_NOT_NULL(client);
-  AZ_PRECONDITION_VALID_SPAN(mqtt_topic, 0, false);
-  AZ_PRECONDITION_NOT_NULL(out_mqtt_topic);
+  AZ_PRECONDITION_NOT_NULL(mqtt_topic);
+  AZ_PRECONDITION(mqtt_topic_size > 0);
 
-  int32_t required_length = az_span_size(str_dps) + az_span_size(str_registrations)
-      + az_span_size(str_put_iotdps_register);
+  az_span mqtt_topic_span = az_span_init((uint8_t*)mqtt_topic, (int32_t)mqtt_topic_size);
+  az_span str_dps_registrations = az_span_slice(str_dps_registrations_res, 0, idx_res_start);
 
-  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic, required_length);
+  int32_t required_length
+      = az_span_size(str_dps_registrations) + az_span_size(str_put_iotdps_register);
 
-  az_span remainder = az_span_copy(mqtt_topic, str_dps);
-  remainder = az_span_copy(remainder, str_registrations);
-  az_span_copy(remainder, str_put_iotdps_register);
+  AZ_RETURN_IF_NOT_ENOUGH_SIZE(mqtt_topic_span, required_length);
 
-  *out_mqtt_topic = az_span_slice(mqtt_topic, 0, required_length);
+  az_span remainder = az_span_copy(mqtt_topic_span, str_dps_registrations);
+  remainder = az_span_copy(remainder, str_put_iotdps_register);
+  remainder = az_span_copy_u8(remainder, '\0');
+
+  if (out_mqtt_topic_length)
+  {
+    *out_mqtt_topic_length = (size_t)required_length;
+  }
 
   return AZ_OK;
 }
 
 // Topic: $dps/registrations/GET/iotdps-get-operationstatus/?$rid=%s&operationId=%s
-AZ_NODISCARD az_result az_iot_provisioning_client_get_operation_status_publish_topic_get(
+AZ_NODISCARD az_result az_iot_provisioning_client_query_status_get_publish_topic(
```

Comment:
```suggestion
AZ_NODISCARD az_result az_iot_provisioning_client_get_query_status_publish_topic(
```

Reply:
In this case I'd like to leave it as-is to clearly show the service supports two separate operations: `register` and `query_status`.